packages/core/lib/util/node-http.ts

```diff
@@ -0,0 +1,110 @@
+// @denoify-ignore
+import type { IncomingMessage, ServerResponse } from "http";
+import type { Http2ServerResponse } from "http2";
+import { Request, ReadableStream } from "cross-undici-fetch";
+import { isAsyncIterable } from "./is-async-iterable";
+
+interface NodeRequest {
+  protocol?: string;
+  hostname?: string;
+  body?: any;
+  url?: string;
+  method?: string;
+  headers: any;
+  req?: IncomingMessage;
+  raw?: IncomingMessage;
+}
+
+export async function getNodeRequest(nodeRequest: NodeRequest): Promise<Request> {
+  const fullUrl = `${nodeRequest.protocol || "http"}://${nodeRequest.hostname || nodeRequest.headers.host || "localhost"}${
+    nodeRequest.url || "/graphql"
+  }`;
+  const maybeParsedBody = nodeRequest.body;
+  const rawRequest = nodeRequest.raw || nodeRequest.req || nodeRequest;
+  if (nodeRequest.method !== "POST") {
+    return new Request(fullUrl, {
+      headers: nodeRequest.headers,
+      method: nodeRequest.method,
+    });
+  } else if (maybeParsedBody) {
+    const request = new Request(fullUrl, {
+      headers: nodeRequest.headers,
+      method: nodeRequest.method,
+    });
+    Object.defineProperties(request, {
+      json: {
+        value: async () => maybeParsedBody,
+      },
+      text: {
+        value: async () => JSON.stringify(maybeParsedBody),
+      },
+      body: {
+        get: () =>
+          new Request(fullUrl, {
+            method: "POST",
+            body: JSON.stringify(maybeParsedBody),
+          }).body,
```

Comment:
`request.clone().body` is gonna be faster, and less code right.

https://developer.mozilla.org/en-US/docs/Web/API/Request/clone

Reply:
This is actually a hack for NodeJS(just to have the same output with other envs). We don't have body in the request on purpose above because it might be `GET` request so there is nothing to clone actually.

Reply:
Why should nodejs be having any impact for consumers of other targets? Apply the "nodejs hacks" for the nodejs target, not for the workers or others

Reply:
This is applied already in a NodeJS helper. Check the name of the function, it is getNodeRequest :) So other consumers don't call this function at all.